pkg/liquidity-mining/contracts/VotingEscrowRemapper.sol

```diff
@@ -0,0 +1,306 @@
+        // - The local address must not be the target remote address for another local user (B --> C is forbidden).
+        //
+        // Note that this means that it is possible to frontrun this call to grief a user by taking up their
+        // selected remote address before they do so. This is mitigated somewhat by restricting potential attackers to
+        // the set of contracts that are allowlisted to hold veBAL (and their remapping managers). Should
+        // one of them grief, then Balancer governance can remove them from these allowlists.
+
+        // Assuming A --> B, then C cannot be remapped to B.
+        // To prevent it, we verify that the reverse mapping of B does not exist.
+        require(
+            _remoteToLocalAddressMap[chainId][remoteUser] == address(0),
+            "Cannot overwrite an existing mapping by another user"
+        );
+
+        // Assuming A --> B, then C cannot be remapped to A.
+        // To prevent it, we verify that the mapping of A does not exist.
+        require(
+            _localToRemoteAddressMap[chainId][remoteUser] == address(0),
+            "Cannot remap to an address that is in use locally"
+        );
+
+        // Assuming A --> B, then B cannot be remapped to C.
+        // To prevent it, we verify that the reverse mapping of B does not exist.
+        require(
+            _remoteToLocalAddressMap[chainId][localUser] == address(0),
+            "Cannot remap to an address that is in use remotely"
+        );
+
+        // This is a best-effort check: we should not allow griefing the existing balance of an account,
+        // because with this remapping we would overwrite it in the target chain ID.
+        require(_votingEscrow.balanceOf(remoteUser) == 0, "Target remote address has non-zero veBAL balance");
+
+        // Clear out the old remote user to avoid orphaned entries.
+        address oldRemoteUser = _localToRemoteAddressMap[chainId][localUser];
+        _remoteToLocalAddressMap[chainId][oldRemoteUser] = address(0);
+
+        // Set up new remapping.
+        _remoteToLocalAddressMap[chainId][remoteUser] = localUser;
+        _localToRemoteAddressMap[chainId][localUser] = remoteUser;
+
+        // Note: it is important to perform the bridge calls _after_ the mappings are settled, since the
+        // omni voting escrow will rely on the correct mappings to bridge the balances.
+        (uint256 nativeFee, ) = omniVotingEscrow.estimateSendUserBalance(uint16(chainId), false, "");
+        if (oldRemoteUser != address(0)) {
+            // If there was an old mapping, send balance from (local) oldRemoteUser --> (remote) oldRemoteUser
+            // This should clean up the existing bridged balance from localUser --> oldRemoteUser.
+            omniVotingEscrow.sendUserBalance{ value: nativeFee }(
+                oldRemoteUser,
+                uint16(chainId),
+                payable(msg.sender),
+                address(0),
+                ""
+            );
```

Comment:
Note for the reviewer: to my best understanding, this is the right way to call the bridge, but let's double check.